### DISCUSSION TITLE
Hi, I want to use imodel.js To achieve roaming along the specified path.

### DISCUSSION THREAD

**Discussion by shenshengyi** [iTwin/itwinjs-core#548] [Q&A] [answered]
Hi, I want to use imodel.js Realize roaming along the specified path. For example, if I draw a path and then roam along the path, can you provide me with an idea? thank you.

**Comment by pmconne** [comment, marked as answer]
Implement a [viewport animator](https://github.com/imodeljs/imodeljs/blob/a7e6b1f650d694642176891b4147122d691c28a3/core/frontend/src/ViewAnimation.ts#L21) and [apply it to a Viewport](https://github.com/imodeljs/imodeljs/blob/a7e6b1f650d694642176891b4147122d691c28a3/core/frontend/src/Viewport.ts#L1941).
In your `Animator`'s `animate` function, interpolate along your path based on the time elapsed to determine where the camera should be positioned and how it should be oriented. You may find functions like [fractionToPoint](https://github.com/imodeljs/imodeljs/blob/a7e6b1f650d694642176891b4147122d691c28a3/core/geometry/src/curve/CurvePrimitive.ts#L97) useful for this. Then adjust the viewport's frustum correspondingly.
Return `true` from `animate` when you reach the end of your path, to terminate the animation.

**Reply by shenshengyi** [reply]
> Implement a [viewport animator](https://github.com/imodeljs/imodeljs/blob/a7e6b1f650d694642176891b4147122d691c28a3/core/frontend/src/ViewAnimation.ts#L21) and [apply it to a Viewport](https://github.com/imodeljs/imodeljs/blob/a7e6b1f650d694642176891b4147122d691c28a3/core/frontend/src/Viewport.ts#L1941).
> In your `Animator`'s `animate` function, interpolate along your path based on the time elapsed to determine where the camera should be positioned and how it should be oriented. You may find functions like [fractionToPoint](https://github.com/imodeljs/imodeljs/blob/a7e6b1f650d694642176891b4147122d691c28a3/core/geometry/src/curve/CurvePrimitive.ts#L97) useful for this. Then adjust the viewport's frustum correspondingly.
> Return `true` from `animate` when you reach the end of your path, to terminate the animation.

Thank you very much for your answer. I'll implement it in this way.
